lib/Support/TimingModels.cpp

```diff
@@ -32,13 +32,20 @@ namespace ljson = llvm::json;
 //===----------------------------------------------------------------------===//
 
 unsigned dynamatic::getOpDatawidth(Operation *op) {
-  // All arithmetic operations are handled the same way
-  if (op->getName().getDialectNamespace() == "arith")
-    return getHandshakeTypeBitWidth(op->getOperand(0).getType());
-
   // Handshake operations have various semantics and must be handled on a
   // case-by-case basis
   return llvm::TypeSwitch<Operation *, unsigned>(op)
+      .Case<handshake::ExtSIOp, handshake::ExtUIOp, handshake::AddFOp,
+            handshake::AddIOp, handshake::AndIOp, handshake::CmpFOp,
+            handshake::CmpIOp, handshake::DivFOp, handshake::DivSIOp,
+            handshake::DivUIOp, handshake::ExtSIOp, handshake::ExtUIOp,
+            handshake::MaximumFOp, handshake::MinimumFOp, handshake::MulFOp,
+            handshake::MulIOp, handshake::NegFOp, handshake::OrIOp,
+            handshake::SelectOp, handshake::ShLIOp, handshake::ShRSIOp,
+            handshake::ShRUIOp, handshake::SubFOp, handshake::SubIOp,
+            handshake::TruncIOp, handshake::XOrIOp>([&](auto) {
```

Comment:
At some point I introduced the `ArithOpInterface` interface that all of these operations implement for the purpose of avoiding these things that become annoying and unscalable (we have to remember to revisit this every time we add a new operation of this kind), so you should just be able to do `Case<handshake::ArithOpInterface>`. However I think `handshake::SelectOp` should have its own case before the interface case; its first operand is always a 1-bit wide condition so you want the width of the second or third operand in that case.